### PR TITLE
added create_user_table.yml, create_user_discount_table.yml, create_l…

### PR DESCRIPTION
**Pull request by MarvelGa**
…ocation_discount_table.yml and added its names to db.changelog-master.yml

### CODE REVIEW

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+databaseChangeLog:
+  - changeSet:
+      id: 1234567890-8
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml]: let's use meaningful ids for changelogs
like this one - "15062021_create_location_discount_table"

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            tableName: location_discount
+            columns:
+              - column:
+                  name: loc_id
+                  type: BIGINT
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                    primaryKeyName: pk_ld_loc_id
+                    foreignKeyName: fk_ld_location
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml]: to improve readability let's use the full names of the tables in fks:
"fk_location_discount_location"

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            schemaName: public
+            tableName: location_discount
+            columns:
+              - column:
+                  name: loc_id
+                  type: BIGINT
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                    primaryKeyName: pk_ld_loc_id
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml]: let's use default names for primary keys by not specifying 'primaryKeyName'

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                    references: location(loc_id)
+              - column:
+                  name: dis_id
+                  type: BIGINT
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                    primaryKeyName: pk_ld_dis_id
+                    foreignKeyName: fk_ld_discount
+                    references: discount(dis_id)
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml:22]: let's add ld_created column to the location_discount table
the value should be precomputed by the db

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                  name: loc_id
+                  type: BIGINT
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                    primaryKeyName: pk_ld_loc_id
+                    foreignKeyName: fk_ld_location
+                    references: location(loc_id)
+              - column:
+                  name: dis_id
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml:18]: add composite primary key for the link tables via separate changeSet

File: src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      changes:
+        - addPrimaryKey:
+            catalogName: discount_web_app
+            clustered: true
+            columnNames: loc_id, dis_id
+            constraintName: pk_location_discount
+            schemaName: public
+            tableName: location_discount
+            validate: true
+            column:
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml]: adding column ld_created should be in the createTable definition

File: src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+databaseChangeLog:
+  - changeSet:
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml]: let's place this changeSet in the same file where we create location_discount table

File: src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+      id: "15062021_add_composite_pk_location_discount_table"
+      author: tesliuk.ua@gmail.com
+      changes:
+        - addPrimaryKey:
+            catalogName: discount_web_app
+            clustered: true
+            columnNames: loc_id, dis_id
+            constraintName: pk_location_discount
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml]: let's name it as pk_loc_id_dis_id

File: src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+databaseChangeLog:
+  - changeSet:
+      id: "15062021_add_composite_pk_location_discount_table"
+      author: tesliuk.ua@gmail.com
+      changes:
+        - addPrimaryKey:
+            catalogName: discount_web_app
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_add_composite_pk_location_discount_table.yml]: do we need to define catalogName here?

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 10]
```diff
@@ -19,3 +19,21 @@ databaseChangeLog:
                   constraints:
                     foreignKeyName: fk_location_discount_discount
                     references: discount(dis_id)
+              - column:
+                  name: id_location_discount_created
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml]: let's rename to ld_created

File: src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml [hunk 11]
```diff
@@ -33,4 +33,24 @@ databaseChangeLog:
                   name: ud_modified
                   type: TIMESTAMP WITHOUT TIME ZONE
                   constraints:
-                    nullable: false
\ No newline at end of file
+                    nullable: false
+              - column:
+                  name: id_user_discount_created
```
Comment (ahlukhouski2) [inline src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml]: ud_created

File: src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+                  type: TIMESTAMP WITHOUT TIME ZONE
+                  constraints:
+                    nullable: false
+              - column:
+                  name: ud_modified
+                  type: TIMESTAMP WITHOUT TIME ZONE
+                  constraints:
+                    nullable: false
+              - column:
+                  name: ud_created
```
Comment (Bozhokmaria) [inline src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml]: ud_created - 2 times

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+                  constraints:
+                   nullable: false
+  - changeSet:
+      id: 15062021_add_composite_pk_location_discount_table
+      author: tesliuk.ua@gmail.com
+      changes:
+        - addPrimaryKey:
+            clustered: true
+            columnNames: loc_id, dis_id
+            constraintName: pk_loc_id_dis_id
```
Comment (dmitriychernov7) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml:37]: rename to pk_location_discount_id

general rule for primary key names is "pk_<table_name>_id"

File: src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+                  constraints:
+                    nullable: false
+  - changeSet:
+      id: 15062021_add_composite_pk_user_discount_table
+      author: tesliuk.ua@gmail.com
+      changes:
+        - addPrimaryKey:
+            clustered: true
+            columnNames: usr_id, dis_id
+            constraintName: pk_usr_id_dis_id
```
Comment (dmitriychernov7) [inline src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml:46]: rename to pk_user_discount_id

File: src/main/resources/db/changelog/init/15062021_create_user_table.yml [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+            tableName: user
+            columns:
+              - column:
+                  name: usr_id
+                  type: BIGINT
+                  autoIncrement: true
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                    primaryKeyName: pk_user
```
Comment (dmitriychernov7) [inline src/main/resources/db/changelog/init/15062021_create_user_table.yml:17]: rename to pk_user_id

File: src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml (hunk 14 above)
Comment (dmitriychernov7) [inline src/main/resources/db/changelog/init/15062021_create_user_discount_table.yml:46]: rename to pk_user_discount

File: src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml (hunk 13 above)
Comment (dmitriychernov7) [inline src/main/resources/db/changelog/init/15062021_create_location_discount_table.yml:37]: rename to pk_location_discount

**Review by ahlukhouski2 — APPROVED** [review]

**Pull request merged**